.github/workflows/changed_solutions.yml

```diff
@@ -0,0 +1,61 @@
+name: Changed Solutions
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.actor, '[bot]') != true }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+          cache: "npm"
+      - run: npm ci
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v22
+      - name: Run solutions in changed files directory
+        run: |
+          root_dir=$(pwd)
+          failed_tests=()
+
+          function run_tests {
+            echo -e "\nRunning test for $1"
+            cd $1
+            npm run test:solutions || failed_tests+=("$1")
+            cd $root_dir
+            return 0
+          }
+
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
```

Comment:
Following the comment about augmenting `solutions.yml`, here's a tricky thing: these projects also rely on some config files from the root directory:

* `package-lock.json`: dependencies such as Jest and TypeScript
* `.prettierrc`: in case formatting changes break.... something, who knows

...and that's about it that I can think of. 

If either of those files change then _all_ projects should be tested.

Reply:
Yea right !. I'll look into merging this into the `solutions.yml`. I think it's better that way !